library/src/main/java/com/wultra/android/mtokensdk/operation/OperationsService.kt

```diff
@@ -163,6 +169,34 @@ class OperationsService: IOperationsService {
         }
     }
 
+    private fun processServerTime(response: OperationListResponse, requestStarted: ZonedDateTime) {
+
+        val now = ZonedDateTime.now()
+        val requestDelayMilliseconds = now.toInstant().toEpochMilli() - requestStarted.toInstant().toEpochMilli()
+
+        // server does not support this feature
+        if (response.currentTimestamp == null) {
+            return
+        }
+
+        // Reject the value if the request took too long and we already have a server date.
+        // This is to avoid volatility of the value
+        if (serverDateShiftInNanoSeconds != null && requestDelayMilliseconds > 1_000) {
+            return
+        }
+
+        // We're adding half of the time that the request took to compensate for the network delay
+        val serverTime = response.currentTimestamp.plusNanos((requestDelayMilliseconds/2) * 1_000_000)
+
+        // If the difference is under 0.3 seconds, we ignore the new value to avoid unnecessary changes that might be due to network delay.
+        val currentServerDate = currentServerDate()
+        if (currentServerDate != null && abs((currentServerDate.toInstant().toEpochMilli() - serverTime.toInstant().toEpochMilli()).toDouble() / 1000) > 0.3) {
```

Comment:
My eyes bleed from these unit conversions.

Reply:
I was just trying if you would catch it. Trust me, bro 🤣